SConstruct

```diff
@@ -638,6 +638,22 @@ print(f'Building for platform "{env["platform"]}", architecture "{env["arch"]}",
 if env.dev_build:
     print("NOTE: Developer build, with debug optimization level and debug symbols (unless overridden).")
 
+# Set x86 CPU instruction sets to use by the compiler's autovectorization.
+if env["arch"] == "x86_64" or env["arch"] == "x86_32":
+    if env.msvc and env["arch"] == "x86_64":
+        # https://stackoverflow.com/questions/64053597/how-do-i-enable-sse4-1-and-sse3-but-not-avx-in-msvc/69328426#69328426
+        env.Append(CCFLAGS=["/d2archSSE42"])
```

Comment:
According to the [CI failure](https://github.com/godotengine/godot/actions/runs/11560485962/job/32177494149?pr=59595), this part is also run under clang-cl even though it doesn't make sense. How can I exclude this specifically under clang-cl?

That said, clang-cl seems to have some kind of MSVC argument compatibility, so maybe it should be fixed upstream by handling `/d2archSSE42` as `-msse4.2`.

Reply:
I am not very sure but in some places I saw following code: `if env.msvc and not using_clang(env)`. (`using_clang` is basically just a check if compiler name contains `clang` substring so I think it should work)